web/src/test/java/org/cbioportal/web/PatientControllerTest.java

```diff
@@ -0,0 +1,243 @@
+            .andExpect(MockMvcResultMatchers.jsonPath("$.cancerStudy.typeOfCancerId")
+                .value(TEST_TYPE_OF_CANCER_ID_1));
+    }
+
+    @Test
+    public void fetchPatientsDefaultProjection() throws Exception {
+
+        List<Patient> patientList = createExamplePatients();
+
+        Mockito.when(patientService.fetchPatients(Mockito.anyListOf(String.class), Mockito.anyListOf(String.class),
+            Mockito.anyString())).thenReturn(patientList);
+
+        List<PatientIdentifier> patientIdentifiers = new ArrayList<>();
+        PatientIdentifier patientIdentifier1 = new PatientIdentifier();
+        patientIdentifier1.setStudyId(TEST_CANCER_STUDY_IDENTIFIER_1);
+        patientIdentifier1.setPatientId(TEST_STABLE_ID_1);
+        patientIdentifiers.add(patientIdentifier1);
+        PatientIdentifier patientIdentifier2 = new PatientIdentifier();
+        patientIdentifier2.setStudyId(TEST_CANCER_STUDY_IDENTIFIER_2);
+        patientIdentifier2.setPatientId(TEST_STABLE_ID_2);
+        patientIdentifiers.add(patientIdentifier2);
+
+        mockMvc.perform(MockMvcRequestBuilders.post("/patients/fetch")
+            .accept(MediaType.APPLICATION_JSON)
+            .contentType(MediaType.APPLICATION_JSON)
+            .content(objectMapper.writeValueAsString(patientIdentifiers)))
+            .andExpect(MockMvcResultMatchers.status().isOk())
+            .andExpect(MockMvcResultMatchers.content().contentType(MediaType.APPLICATION_JSON))
+            .andExpect(MockMvcResultMatchers.jsonPath("$", Matchers.hasSize(2)))
+            .andExpect(MockMvcResultMatchers.jsonPath("$[0].internalId").doesNotExist())
+            .andExpect(MockMvcResultMatchers.jsonPath("$[0].stableId").value(TEST_STABLE_ID_1))
+            .andExpect(MockMvcResultMatchers.jsonPath("$[0].cancerStudyId").value(TEST_CANCER_STUDY_IDENTIFIER_1))
+            .andExpect(MockMvcResultMatchers.jsonPath("$[0].cancerStudyIdentifier").doesNotExist())
+            .andExpect(MockMvcResultMatchers.jsonPath("$[0].cancerStudy").doesNotExist())
+            .andExpect(MockMvcResultMatchers.jsonPath("$[1].internalId").doesNotExist())
+            .andExpect(MockMvcResultMatchers.jsonPath("$[1].stableId").value(TEST_STABLE_ID_2))
+            .andExpect(MockMvcResultMatchers.jsonPath("$[1].cancerStudyId").value(TEST_CANCER_STUDY_IDENTIFIER_2))
+            .andExpect(MockMvcResultMatchers.jsonPath("$[1].cancerStudyIdentifier").doesNotExist())
+            .andExpect(MockMvcResultMatchers.jsonPath("$[1].cancerStudy").doesNotExist());
+    }
+
+    @Test
+    public void fetchPatientsMetaProjection() throws Exception {
+
+        BaseMeta baseMeta = new BaseMeta();
+        baseMeta.setTotalCount(2);
+
+        Mockito.when(patientService.fetchMetaPatients(Mockito.anyListOf(String.class),
+            Mockito.anyListOf(String.class))).thenReturn(baseMeta);
+
+        List<PatientIdentifier> patientIdentifiers = new ArrayList<>();
```

Comment:
here is reuse opportunity for construction code .. why not externalize it? (not necessary .. up to you)